### PR TITLE
sql: allow real/float4 to be used

### PR DESCRIPTION
**Pull request by sploiselle**
Previously, `real` were largely (totally?) inaccessible to users because expressing it in SQL always got converted to `double`. This PR fixes that and adds some missing casts.

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/4918)

### CODE REVIEW

File: test/testdrive/float.td [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  FORMAT AVRO USING SCHEMA '${schema}' ENVELOPE DEBEZIUM
+
+> CREATE MATERIALIZED VIEW data_view as SELECT * from data
+
+> SELECT * FROM data_view
+1.2
+18000000000000000000000000000000000000000
+
+! SELECT a::float4 FROM data_view
+float out of range
```
Comment (benesch) [inline test/testdrive/float.td:58]: Just mentioned this in Slack, but you should be able to input these floats exactly by using the string literal syntax (`'1.2'::float4`). If that does work, can we make these SLTs instead? 🤞

**Review by benesch — APPROVED** [review]
Nice catch!

File: test/testdrive/float.td (hunk 1 above)
Comment (sploiselle) [reply test/testdrive/float.td:58]: Yep! Will do that in just a sec. ty for the pointer.

**Review by benesch — APPROVED** [review]
🚢

**Pull request merged**